shm/Map.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include "Island.hpp"
+#include <vector>
+
+class Map {
+private:
```

Comment:
I think we should decide on particular style of private/public/protected blocks of code. In Cargo class I've done public first, as it was in example solution from the lesson. Do you know what are good/bad practices in this case?

I found this very intuitive, but it's just stackoverflow: [https://stackoverflow.com/a/1760867/8964273](url)

> Personally, I prefer putting public methods first, followed by protected methods, following by private methods. Member data should in general always be private or protected, unless you have a good reason for it not to be so.
> 
> My rationale for putting public methods at the top is that it defines the interface for your class, so anyone perusing your header file should be able to see this information immediately.
> 
> In general, private and protected members are less important to most people looking at the header file, unless they are considering modifying the internals of the class. Keeping them "out of the way" ensures this information is maintained only on a need to know basis, one of the more important aspects of encapsulation.

Reply:
TBH I do private fields first because usually I know what private members class will have at the beginning and  they are needed later for public methods etc. It's not very intuitive for me to write declaration of public getter for member `size_` when private variable `size_` wasn't even defined yet